### PR TITLE
Send actual path within EventData

### PR DESCRIPTION
**Pull request by alessandroraffa**
Also fixed some missing type declarations

### CODE REVIEW

File: src/SerialPort.ts [hunk 1]
```diff
@@ -64,7 +65,7 @@ class SerialPort {
    * @param {string} hex the hex of data
    * @returns {Promise} success promise
    */
-  send(hex: string) {
+  send(hex: string): boolean {
```
Comment (bastengao) [inline src/SerialPort.ts:68]:
```suggestion
  send(hex: string): Promise<boolean> {
```

File: src/SerialPort.ts [hunk 2]
```diff
@@ -73,14 +74,17 @@ class SerialPort {
    * @param {listener} listener
    * @returns {Subscription} subscription
    */
-  onReceived(listener: (data: Buffer) => void): EventSubscription {
+  onReceived(listener: (buffer: Buffer, path: string) => void): EventSubscription {
```
Comment (bastengao) [inline src/SerialPort.ts:77]: I think it's not necessary to receive path.
```suggestion
  onReceived(listener: (buffer: Buffer) => void): EventSubscription {
```

File: src/SerialPort.ts [hunk 3]
```diff
@@ -73,14 +74,17 @@ class SerialPort {
...
+      if (!event.path || this.path !== event.path) {
+        return;
+      }
       if (!event.data) {
         return;
       }
 
       const buff = Buffer.from(event.data, 'hex');
-      listener(buff);
+      listener(buff, event.path);
```
Comment (bastengao) [inline src/SerialPort.ts:87]:
```suggestion
      listener(buff);
```

File: src/SerialPort.ts (hunk 1 above)
Comment (alessandroraffa) [reply src/SerialPort.ts:68]: Is that a Promise?

**Comment by alessandroraffa** [comment]
We added the `path` in the listener as a debug value for our app.
But actually, if the data is filtered, it's useless.

File: src/SerialPort.ts (hunk 1 above)
Comment (alessandroraffa) [reply src/SerialPort.ts:68]: Yes, I've missed the Java implementation. It's a promise.

**Pull request merged**